src/freenet/clients/http/SecurityLevelsToadlet.java

```diff
@@ -686,6 +693,25 @@ public static void generatePasswordFormPage(boolean wasWrong, ToadletContainer c
 		HTMLNode form = ctx.addFormChild(content, postTo, "masterPasswordForm");
 		generatePasswordFormPage(wasWrong, form, content, forDowngrade, forUpgrade, physicalSecurityLevel, redirect);
 	}
+	
+	public static void generatePasswordConfirmationForm(HTMLNode formNode) {
```

Comment:
@bertm A lot of the methods in this class that draw components are static and therefor can only call other static methods. I have changed it to private but I was wondering if you would like me to change all of the other methods so they are not static?